test/framework/framework.go

```diff
@@ -45,8 +45,11 @@ import (
 
 const (
 	admissionHookSecretName                 = "prometheus-operator-admission"
```

Comment:
do we still need this secret?

Reply:
I suppose not since the tests are leveraging the standalone server now. Will clean up

Reply:
Actually we have a test depending on this, and since we intend, for now at least to continue to support TLS in the operator I vote we keep this and related code, wdyt?

https://github.com/prometheus-operator/prometheus-operator/blob/main/test/e2e/main_test.go#L297

Reply:
Alright, let's keep as-is for now. IIUC we have only one [e2e test](https://github.com/prometheus-operator/prometheus-operator/blob/023feeca9d01be2fb114e0a6b7ffa67a5928de08/test/e2e/alertmanager_test.go#L1070-L1101) exercising the validation logic (it's buried into a more generic test and only applies to AlertmanagerConfig). As we're splitting the functionality, I would expect that we have dedicated e2e tests for the webhook service (it can be a follow-up PR). WDYT?

Reply:
right, my goal here was to essentially swop out the two implementations and get CI green when doing so, to ensure we maintained what we have and weren't breaking anything by doing so.

And I believe you are correct. So what the e2e tests (webhook related) currently achieve is this:

1. Exercise the manifests generated via jsonnet and ensure the pods come up
2. Cover some cases of AlertmanagerConfig validation as you mentioned

We do however have some integration tests for the server https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/admission/admission_test.go which cover a good chunk.

So I feel like we are in reasonable shape with this. Happy to do a follow up PR to close out any gaps. I'm not sure we should duplicate everything as an e2e however. 

The way we have structured `AlertManagerConfig` validation at least is that the webhook calls out to exposed function https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/alertmanager/validation.go#L35 which is also fairly well unit tested.

Again, not against the idea of spending more time on e2e tests for this area, just wanted to document what we have right now.

Reply:
> So I feel like we are in reasonable shape with this. Happy to do a follow up PR to close out any gaps. I'm not sure we should duplicate everything as an e2e however.

I think that the webhook validation binary would deserve some e2e testing to ensure that we don't regress with newer Kubernetes versions. I wouldn't duplicate everything but one happy case/one sad case per resource sounds reasonable.
Having it as a follow-up task is perfectly fine by me.